src/result-async.ts

```diff
@@ -30,8 +30,8 @@ export class ResultAsync<T, E> {
     return new ResultAsync(newPromise)
   }
 
-  map = <A>(f: (t: T) => A | Promise<A>) =>
-    new ResultAsync(
+  map<A>(f: (t: T) => A | Promise<A>) {
```

Comment:
While we're at it, would you be able to also specify the return type for all of these methods?

Reply:
I believe I removed them at your request because the compiler could extrapolate them. Maybe I misunderstood.

Is it necessary if the compiler is capable of extrapolating them ?

Reply:
Hmm I don't believe I requested to remove the return type. If I did I apologise.

like you mentioned, It's not necessary as the compiler can infer the return type. But by having an explicit type annotation, you create a contract. And in the future if another person comes in and changes this code, then it must adhere to the contract! And we can tell if the change still adheres to the contract or not via typechecking.

Reply:
Ok thanks for clearing this up. I just pushed a commit to add the return types back.